### PR TITLE
Third take to convert jekyll's posts to cobalt

### PR DESCRIPTION
**Pull request by Geobert**
Had to delete my fork and start again. I don't know how to work with git and forks

### CODE REVIEW

**Comment by epage** [comment]
My workflow
```
git clone my repo
git remote add upstream https://github.com/cobalt-org/cobalt.rs.git
git checkout -b myfeature  # master is reserved for tracking upstream
... work
# Prepare to push to github
git checkout master
git pull upstream master
git checkout myfeature
git rebase master  # usually comes out clean.  Resolving conflicts can be a bit confusing
git push
```

**Comment by epage** [comment]
You renamed the key to access the value by but did not rename the flag.

btw when you do this, do you mind also squashing your commits so we have a cleaner history.

```
git rebase -i HEAD~5
```
will bring up your selected editor (WARNING: default is vim) with the last 5 commits.  You edit what action is to be performed (switch pick to squash or fix), save, quite, and your history is edited.

**Comment by epage** [comment]
Thanks for your work on this!

File: src/jekyll/jk_document.rs [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+
+fn split_document(content: &str) -> Result<(Option<&str>, &str)> {
+    if FRONT_MATTER_DIVIDE.is_match(content) {
+        let mut splits = FRONT_MATTER_DIVIDE.splitn(content, 3);
+        let first = splits.next().unwrap_or("");
+        let second = splits.next().unwrap_or("");
+        let third = splits.next().unwrap_or("");
+
+        if third.is_empty() {
+            // only one "---"
```
Comment (epage) [inline src/jekyll/jk_document.rs]: Is there a reason you ended up supporting this use case?  Does jekyll actually support it?

Comment (Geobert) [reply src/jekyll/jk_document.rs]: I don't know ^^' I'm fairly new to the world of static web site generator. I wanted to be robust here. I'll try to find some info and maybe find out that an error is more appropriate :)

Comment (epage) [reply src/jekyll/jk_document.rs]: Thanks.

Comment (Geobert) [reply src/jekyll/jk_document.rs]: If I believe Jekyll's documentation on frontmatter https://jekyllrb.com/docs/frontmatter/

> The front matter is where Jekyll starts to get really cool. Any file that contains a YAML front matter block will be processed by Jekyll as a special file. The front matter must be the first thing in the file and must take the form of valid YAML set between triple-dashed lines.

So it seems that I don't need to manage this but spit an error instead :)

Comment (epage) [reply src/jekyll/jk_document.rs]: Already made the change in #298 as well as fix some bugs I forgot about during the review.

**Pull request merged**